includes/api/class-wc-rest-system-status-controller.php

```diff
@@ -0,0 +1,750 @@
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'system-status';
+
+	/**
+	 * Register the routes for coupons.
+	 */
+	public function register_routes() {
+        register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+                'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+    /**
+	 * Check whether a given request has permission to view system status
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+        if ( ! wc_rest_check_manager_permissions( 'system-status', 'read' ) ) {
+        	return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot view.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		return true;
+	}
+
+    /**
+	 * Get a system status info, by section.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$schema    = $this->get_item_schema();
+		$mappings  = $this->get_item_mappings();
+		$response  = array();
+
+		foreach ( $mappings as $section => $values ) {
+			settype( $values, $schema['properties'][ $section ]['type'] );
+			foreach( $values as $key => $value ) {
+				if ( isset( $schema['properties'][ $section ]['properties'][ $key ]['type'] ) ) {
+					settype( $values[ $key ], $schema['properties'][ $section ]['properties'][ $key ]['type'] );
+				}
+			}
+			$response[ $section ] = $values;
+		}
+
+		return rest_ensure_response( $response );
+	}
+
+    /**
+	 * Get the system status schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'system-status',
+			'type'       => 'object',
+			'properties' => array(
+				'environment' => array(
+					'description' => __( 'Environment', 'woocommerce' ),
+					'type'        => 'array',
+					'context'     => array( 'view', 'edit' ),
+					'properties'  => array(
+						'home_url' => array(
+							'description' => __( 'Home URL' ),
+							'type'        => 'string',
+		                    'format'      => 'uri',
+							'context'     => array( 'view', 'edit' ),
+						),
+		                'site_url' => array(
+		                    'description' => __( 'Site URL' ),
+		                    'type'        => 'string',
+		                    'format'      => 'uri',
+		                    'context'     => array( 'view', 'edit' ),
+		                ),
+		                'wc_version' => array(
```

Comment:
Note sure if we need a prefix here, works only as `version` since you are checking the system status of a WooCommerce store.